src/write_gcm.c

```diff
@@ -2147,6 +2176,53 @@ static monitored_resource_t *wg_monitored_resource_create(
   return NULL;
 }
 
+static monitored_resource_t *wg_monitored_resource_create_from_metadata(
+    const char *metadata_agent_url, const char *metadata_agent_port,
+    const char *project_id, const char *host) {
+  if (host == NULL) {
+    WARNING("write_gcm: Hostname not set. Cannot use Metadata Agent.");
+    return NULL;
+  }
+  char *body = NULL;
+  const char **headers = NULL;
+  int num_headers = 0;
+  const char *url = metadata_agent_url != NULL ?
+      metadata_agent_url : metadata_agent_default_url;
+  const char *port = metadata_agent_port != NULL ?
+      metadata_agent_port : metadata_agent_default_port;
+  char *response_buffer = (char *)
+      calloc(METADATA_RESPONSE_BUFFER_SIZE, sizeof(char));
+  if (response_buffer == NULL) {
+    ERROR(
+        "write_gcm: wg_monitored_resource_create_from_metadata: calloc failed.");
+    return NULL;
+  }
+  size_t url_size = strlen(url) + strlen(port) + strlen(host) + 21;
+  {
+  char query[url_size];
+  int written = ssnprintf(query, url_size, "%s:%s/monitoredResource/%s", url,
+      port, host);
+  DEBUG("write_gcm: wg_monitored_resource_create_from_metadata: "
+        "Written: %d, Size: %zu, URL: %s", written, url_size, query);
+
+  // ssnprintf returns number of characters written excluding the \0 character
+  // but the size argument it takes requires \0 to be accounted for.
+  if (url_size - written != 1) {
+    ERROR("write_gcm: wg_monitored_resource_create_from_metadata: %s",
+          "Could not create Metadata URL.");
```

Comment:
Let's be clear on which metadata we're talking about. How about `metadata agent URL`?